modules/building-memcached-operator-using-osdk.adoc

```diff
@@ -0,0 +1,256 @@
+$ kubectl create -f deploy/operator.yaml
+----
+
+.. Verify that the `memcached-operator` pod is up and running:
++
+----
+$ kubectl get pods
+NAME                                  READY     STATUS    RESTARTS   AGE
+memcached-operator-75c4b4c665-8jnj5   1/1       Running   0          20s
+----
+
+. To verify that the Operator can deploy a Memcached application,
+create a `Memcached` custom resource (CR).
+
+.. Modify the *_deploy/cr.yaml_* file as shown:
++
+----
+$ cat deploy/cr.yaml
+apiVersion: "cache.example.com/v1alpha1"
+kind: "Memcached"
+metadata:
+  name: "example-memcached"
+spec:
+  size: 3
+----
+
+.. Create the `Memcached` CR:
++
+----
+$ kubectl apply -f deploy/cr.yaml
+----
+
+.. Verify that the Memcached Operator creates the deployment for the CR:
++
+----
+$ kubectl get deployment
+
+NAME                     DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+memcached-operator       1         1         1            1           2m
+example-memcached        3         3         3            3           1m
+----
+
+.. Check the pods and CR status to confirm the status is updated with the
+`memcached` pod names:
++
+----
+$ kubectl get pods
+
+NAME                                  READY     STATUS    RESTARTS   AGE
+example-memcached-6fd7c98d8-7dqdr     1/1       Running   0          1m
+example-memcached-6fd7c98d8-g5k7v     1/1       Running   0          1m
+example-memcached-6fd7c98d8-m7vn7     1/1       Running   0          1m
+memcached-operator-7cc7cfdf86-vvjqk   1/1       Running   0          2m
+----
++
+----
+$ kubectl get memcached/example-memcached -o yaml
+
+apiVersion: cache.example.com/v1alpha1
+kind: Memcached
+metadata:
+  clusterName: ""
+  creationTimestamp: 2018-03-31T22:51:08Z
+  generation: 0
+  name: example-memcached
+  namespace: default
+  resourceVersion: "245453"
+  selfLink: /apis/cache.example.com/v1alpha1/namespaces/default/memcacheds/example-memcached
+  uid: 0026cc97-3536-11e8-bd83-0800274106a1
+spec:
+  size: 3
+status:
+  nodes:
+  - example-memcached-6fd7c98d8-7dqdr
+  - example-memcached-6fd7c98d8-g5k7v
+  - example-memcached-6fd7c98d8-m7vn7
+----
+
+. To verify that the Operator can manage a deployed Memcached application, update
+the size of the deployment.
+
+.. Change the `spec.size` field in the `memcached` CR from 3 to 4:
++
+----
+$ cat deploy/cr.yaml
+apiVersion: "cache.example.com/v1alpha1"
+kind: "Memcached"
+metadata:
+  name: "example-memcached"
+spec:
+  size: 4
+----
+
+.. Apply the change:
++
+----
+$ kubectl apply -f deploy/cr.yaml
+----
+
+.. Confirm that the Memcache Operator changes the deployment size:
++
+----
+$ kubectl get deployment
+
+NAME                 DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+example-memcached    4         4         4            4           5m
+----
+
+. Finally, clean everything up:
++
+----
+$ kubectl delete -f deploy/operator.yaml
+$ kubectl delete -f deploy/rbac.yaml
+$ kubectl delete -f deploy/crd.yaml
+$ kubectl delete -f deploy/cr.yaml
```

Comment:
The deleting order is incorrect. Should delete the `cr` before deleting the `crd`, as below:
```
￼$ kubectl delete -f deploy/cr.yaml
$ kubectl delete -f deploy/crd.yaml
```